gammapy/modeling/tests/test_parameter.py

```diff
@@ -369,3 +375,33 @@ def test_priorparameters_to_table(priorpars):
     assert len(table.columns) == 7
     assert table["name"][0] == "spam"
     assert table["value"][1] == 99
+
+
+def test_parameter_set_min_max_error(default_parameter):
+    par = default_parameter
+    assert_equal(par.min, np.nan)
+    assert_equal(par.max, np.nan)
+
+    par.set_lim(min=1, max=10)
+    assert par.min == 1
+    assert par.max == 10
+
+    par.set_lim(max=5)
+    assert par.max == 5
+    assert par.min == 1
+
+    par.set_lim((2, 4))
+    assert par.min == 2
+    assert par.max == 4
+
+    assert par.error == 0
+
+
+def test_set_quantity_str_float(default_parameter):
+    par = default_parameter
+    assert par._set_quantity_str_float(1) == 1
+    assert par._set_quantity_str_float("3 TeV") == 3
+    assert par._set_quantity_str_float(2 * u.TeV) == 2
+    assert par._set_quantity_str_float(4e3 * u.GeV) == 4
```

Comment:
Add asserts on `par.error` when units are passed.

What happens when one of the parameters has units, the other not, eg `par.set_lim(min=1, max=1*u.GeV)`?
Is the error message reasonable?

Reply:
It does not raise any error. Should it ? It is easy to implement if we want it.